### DISCUSSION TITLE
How to disable antialiasing in rendered image?

### DISCUSSION THREAD

**Discussion by csabaj** [pygfx/pygfx#640] [Q&A] [answered]
I've been trying to completely disable antialiasing in pygfx, without much success. The problem is illustrated by running the [wgpu-py](https://github.com/pygfx/wgpu-py/blob/main/examples/triangle.py) and the [pygfx](https://github.com/pygfx/pygfx/blob/main/examples/introductory/hello_triangle.py) hello-triangle examples, producing different behaviour:

wgpu-py:
![webgpu-py_triangle_shade](https://github.com/pygfx/pygfx/assets/159151736/cc3d7316-a6e3-4111-80d7-0c8223dda833)
pygfx:
![pygfx_triangle_shade](https://github.com/pygfx/pygfx/assets/159151736/1e109ac8-caa6-4770-8d97-afd0bee845f0)

I would like to reproduce the wgpu-py behaviour by modifying the pygfx example, could anyone help me on how to achieve that?

**Comment by almarklein** [comment]
I think `renderer.pixel_ratio = 1` should do the trick.

Context: Pygfx renderes to an offscreen texture that (if you have a non-hidpi screen) is twice as large. When rendering that (larger) texture to screen, the data is smoothed. This is known as FSAA (full-screen antialiasing).

**Comment by almarklein** [comment, marked as answer]
Oh, I just realized that even with it set to 1, it will perform a wee bit of smoothing, so it will produce a different result. As it happens, I have an open PR #638 that will allow turning off all smoothing in the final step.

**Reply by csabaj** [reply]
Thanks a lot for the very quick answer! Using your proposed change reproduces the behaviour I was after! FYI, documenting this option to turn off filtering once the PR is approved would be great. It is useful if you use the package for scientific computing where you want to do further computation with the rasterized image's pixel values.
